crates/codegen/ebnf/src/nodes.rs

```diff
@@ -0,0 +1,121 @@
+pub enum EbnfNode {
+    Alternatives {
```

Comment:
Why start using different naming i.e. `Alternatives` rather than `Choice`and `Repeat` rather than `Repeated
`?

Reply:
This is intended to match EBNF specific terminology. Different generators will convert our schema-specific nodes like `choices` into EBNF-specific nodes like `alternatives`: https://en.wikipedia.org/wiki/Extended_Backus%E2%80%93Naur_form#Table_of_symbols

However, this is not entirely accurate (yet). As I'm tackling #123 next, I will need to decide on a language (EBNF/ABNF/others?) and change `EbnfNode` to match. The current one here is just a temporary step.
Maybe I should add this in a comment?

Reply:
My concern is that our grammar uses ordered choice, not traditional alternatives, and so this might be misleading. Also I'm not seeing where metadata such as precedence is handled. Overall I think we shouldn't be so determined to use a formality 100% formally specified. EBNF et al really haven't kept up with parsing techniques (being an invention of the 60s), and are especially not suited to recursive descent or operator precedence parsing. And I strongly think we should prioritise our implementation strategy over e.g. *BNF compliance.

But not blocking for me.